scipy/linalg/tests/test_decomp_cholesky.py

```diff
@@ -266,3 +266,116 @@ def test_empty(self, dt):
+
+
+class TestCholeskyUpdate:
+    def test_update(self):
+        # Test rank-1 update
+        n = 1000
+        R, z = gen_input(n)
+        U = cholesky_update(R, z)
+        tol = n * np.spacing(100.)
+        assert_allclose(U.T.dot(U) - (R.T.dot(R) + np.outer(z, z)),
+                        np.zeros([n, n]), atol=tol)
+
+    def test_downdate(self):
+        # Test rank-1 downdate
+        n = 1000
+        R, z = gen_input(n)
+        D = cholesky_update(R, z, downdate=True)
+        tol = n * np.spacing(100.)
+        assert_allclose(D.T.dot(D) - (R.T.dot(R) - np.outer(z, z)),
+                        np.zeros([n, n]), atol=tol)
+
+    def test_list_input(self):
+        # Test list as input
+        n = 1000
+        R, z = gen_input(n)
+        D = cholesky_update(list(R), list(z))
+        assert_allclose(D.T.dot(D), (R.T.dot(R) + np.outer(z, z)))
+
+    @pytest.mark.parametrize('bad_value', [np.nan, np.inf])
+    def test_finite(self, bad_value):
+        # Test with nan and inf in the input
+        n = 10
+        R0, z = gen_input(n)
+        message = 'array must not contain infs or NaNs'
+
+        R = R0.copy()
+        R[0, 0] = bad_value
+        with pytest.raises(ValueError, match=message):
+            cholesky_update(R, z)
+
+        z[0] = bad_value
+        with pytest.raises(ValueError, match=message):
+            cholesky_update(R0, z)
+
+    def test_dimensions(self):
+        # Tests related to dimension of both array and vector
+        n = 10
+        R, z = gen_input(n)
+
+        message = "Expected 2D array to be updated."
+        with pytest.raises(ValueError, match=message):  # too many dims in R
+            cholesky_update(np.expand_dims(R, 0), z)
+        with pytest.raises(ValueError, match=message):  # too few dims in R
+            cholesky_update(z, z)
+
+        message = "Expected 1D update vector."
+        with pytest.raises(ValueError, match=message):
+            cholesky_update(R, R)
+
+        message = "Input needs to be a square matrix."
+        with pytest.raises(ValueError, match=message):
+            cholesky_update(R[:, :-1], z)
+
+        message = "Input z has to have same length as the number of rows as input R."
+        with pytest.raises(ValueError, match=message):
+            cholesky_update(R[:-1, :-1], z)
+
+    def test_param_overwrite(self):
+        # Test param overwrite
+        n = 100
+
+        # Test with overwrite_R=False and overwrite_z=False
+        R, z = gen_input(n)
+        R_copy = R.copy()
+        z_copy = z.copy()
+        cholesky_update(R, z)
+        assert_equal(R, R_copy)
+        assert_equal(z, z_copy)
+
+        # Test with overwrite_R=True and overwrite_z=True
+        R, z = gen_input(n)
+        R_copy = R.copy()
+        z_copy = z.copy()
+        cholesky_update(R, z, overwrite_R=True, overwrite_z=True)
+        assert_raises(AssertionError, assert_equal, R, R_copy)
+        assert_raises(AssertionError, assert_equal, z, z_copy)
```

Comment:
Maybe let's check that the output array `is` the input array, if that works, or that they are equal otherwise. Also, let's just `assert (R != R_copy).all()` if that passes, and `any` otherwise.